### ISSUE TITLE
[FEA] Enable `.str.find_multiple` public API

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#10126]
**Is your feature request related to a problem? Please describe.**
We currently have `cudf._lib.strings.find_multiple` which is an internal API. We would like to expose it via a public API similar to what we wanted to do in: https://github.com/rapidsai/cudf/pull/4575, this was previously dropped due to lack of ListColumn support: https://github.com/rapidsai/cudf/issues/4569#issuecomment-600807535, which shouldn't be a blocker now.

**Comment by bdice** [comment]
Additional context (@galipremsagar beat me to filing an issue! 😉):

The gpu-bdb query 18 code uses this private API, `cudf._lib.strings.find_multiple`, to search a column of string values for a list of target values:
https://github.com/rapidsai/gpu-bdb/blob/f67ea68d4b422c998d0284cc69fc82ed371c69cc/gpu_bdb/queries/q18/gpu_bdb_query_18.py#L106-L126

**Describe the solution you'd like**
Query 18 should be able to call a public API `Series.str.find_multiple` that behaves like [Series.str.find](https://docs.rapids.ai/api/cudf/nightly/api_docs/api/cudf.core.column.string.StringMethods.find.html#cudf.core.column.string.StringMethods.find), rather than a private API. The implementation in libcudf can be refactored to return a list column of integers (previously proposed by @davidwendt here: https://github.com/rapidsai/cudf/issues/6341#issuecomment-780632145).

@VibhuJawa noted that this would be helpful for a variety of natural language processing tasks.

**Comment by davidwendt** [comment]
Technically you can build the lists column in the Python layer by just creating an offsets column an using it along with column returned by the `find_multiple` API.

```
s = cudf.Series([ "abc", "bcd", "c", "de" ])
result = s.str.find_multiple(["c", "d"])
# result is now [ 2, -1,   1, 2   0, -1   -1, 0 ]

offsets = [ 0, 2, 4, 6, 8 ]  # just an even sequence  [0 : len(s)*len(r) : len(r)]
# likely there is a numpy way to create this array
```
Build a ListColumn by using `offsets` and `results` as its child columns.

**Comment by bdice** [comment]
This API is also a candidate for a separate NLP-focused submodule (which has been discussed among cudf developers a few times but I am not aware of an issue tracking this idea). The pandas API does not have an equivalent of `Series.str.find_multiple`.

**Comment by GregoryKimball** [comment]
Closed by #10134

**Comment by bdice** [comment]
Re-opening this one because it’s only partially complete. The proposed Python API has not yet been added. See: https://github.com/rapidsai/cudf/pull/10134#pullrequestreview-863880065
